L1Trigger/L1TTrackMatch/plugins/L1TrackJetProducer.cc

```diff
@@ -142,509 +143,237 @@ L1TrackJetProducer::~L1TrackJetProducer() {}
 void L1TrackJetProducer::produce(Event &iEvent, const EventSetup &iSetup) {
   unique_ptr<TkJetCollection> L1L1TrackJetProducer(new TkJetCollection);
 
-  // For TTStubs
-  const TrackerTopology &tTopo = iSetup.getData(tTopoToken_);
+  // Read inputs
+  ESHandle<TrackerTopology> tTopoHandle;
+  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
+  ESHandle<TrackerGeometry> tGeomHandle;
+  iSetup.get<TrackerDigiGeometryRecord>().get(tGeomHandle);
+  const TrackerTopology *const tTopo = tTopoHandle.product();
 
   edm::Handle<vector<TTTrack<Ref_Phase2TrackerDigi_> > > TTTrackHandle;
   iEvent.getByToken(trackToken_, TTTrackHandle);
-  vector<TTTrack<Ref_Phase2TrackerDigi_> >::const_iterator iterL1Track;
 
-  edm::Handle<std::vector<l1t::Vertex> > PVtx;
+  edm::Handle< std::vector<l1t::Vertex> > PVtx;
   iEvent.getByToken(PVtxToken_, PVtx);
-  float PVz = (PVtx->at(0)).z0();
+  float PVz = (PVtx->at(0)).z0(); 
 
   L1TrkPtrs_.clear();
-  zBinCount_.clear();
   tdtrk_.clear();
-  unsigned int this_l1track = 0;
-  for (iterL1Track = TTTrackHandle->begin(); iterL1Track != TTTrackHandle->end(); iterL1Track++) {
+
+  for (unsigned int this_l1track=0; this_l1track<TTTrackHandle->size(); this_l1track++){
     edm::Ptr<L1TTTrackType> trkPtr(TTTrackHandle, this_l1track);
-    this_l1track++;
     float trk_pt = trkPtr->momentum().perp();
     int trk_nstubs = (int)trkPtr->getStubRefs().size();
     float trk_chi2dof = trkPtr->chi2Red();
     float trk_d0 = trkPtr->d0();
     float trk_bendchi2 = trkPtr->stubPtConsistency();
-    float trk_z0 = trkPtr->z0();
 
-    int trk_nPS = 0;
+    int trk_nPS = 0; 
     for (int istub = 0; istub < trk_nstubs; istub++) {  // loop over the stubs
       DetId detId(trkPtr->getStubRefs().at(istub)->getDetId());
       if (detId.det() == DetId::Detector::Tracker) {
-        if ((detId.subdetId() == StripSubdetector::TOB && tTopo.tobLayer(detId) <= 3) ||
-            (detId.subdetId() == StripSubdetector::TID && tTopo.tidRing(detId) <= 9))
+        if ((detId.subdetId() == StripSubdetector::TOB && tTopo->tobLayer(detId) <= 3) ||
+            (detId.subdetId() == StripSubdetector::TID && tTopo->tidRing(detId) <= 9))
           trk_nPS++;
       }
     }
+    // select tracks
     if (trk_nPS < trkNPSStubMin_)
-      continue;
+      continue; 
     if (!trackQualityCuts(trk_nstubs, trk_chi2dof, trk_bendchi2))
       continue;
-    if (std::abs(PVz - trk_z0) > dzPVTrk_)
+    if ( fabs(PVz-trkPtr->z0()) > dzPVTrk_ && dzPVTrk_>0 )
```

Comment:
You should keep std::abs and not switch to fabs (CMSSW recommendation)

Reply:
thank you, changed